python/sdss_access/sync/curl.py

```diff
@@ -0,0 +1,285 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+# The line above will help with 2to3 support.
+
+from os.path import isfile, exists, dirname, join, basename, getsize, getmtime, sep
+from re import search
+from sdss_access import SDSSPath, AccessError
+from sdss_access.sync.auth import Auth
+from sdss_access.sync.stream import Stream
+import re
+from sdss_access import os_windows
```

Comment:
General comment here.  I think `sdss_access` should provide a single class, e.g. `Sync` or 'Access' that behaves exactly as `RsyncAccess` or `CurlAccess`, and determines whether an OS is Windows or Posix and returns either `RsyncAccess` or `CurlAccess`.  This way all other software that uses sdss_access has a single entry point, and let's `sdss_access` do the work of OS determination.  Right now all other software would have to do this themselves.  I think you can do with easily with the `__new__` method.  I'm happy to help with this.

Reply:
I am ok with this change, but I thought it would be a feature if a non-windows user had the choice to use rsync or curl (or globus) depending on their preference instead of the code deciding for you.

Reply:
I think that flexibility would still be there.  People can still use `sdss_access` as they do now, e.g. `from sdss_access.sync import RsyncAccess`.  Similarly with `CurlAccess`.  This options just allows one the option to import a smart handler for cases where they need to build cross-platform tools.

Reply:
Agreed.

Reply:
I'm not sure we did the Access class right, but it is really simple, and it does work.  @havok2063  please check if its ok, or too hacky.

Reply:
It's a bit different but I think it will work just fine.